### PR TITLE
Fixes #9615 - Upgrade to Htmlsanitizer v5

### LINKED ISSUE #9615: Update HtmlSanitizer
As one that's been using AngleSharp for quite a while on earlier versions of Umbraco 8 (8.6.4), i was a bit confused when trying to upgrade to 8.10.1 today.

Seeing that it has a dependency on a 2 year old version of HtmlSanitizer, and that in turn having a dependency on a 3 year old version of AngleSharp, stopping me from upgrading, unless i downgrade AngleSharp.

Any reason for these old versions suddenly appearing, or would it be possible to include a newer version of HtmlSanitizer, and by that a newer AngleSharp?

### PR DESCRIPTION
**Pull request by nzdev**
### Prerequisites

- [ ] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes 

### Description
Restores HtmlSanitizer v5. Builds. Also some small optimisations for building the icon lists. IHtmlSanitizer injected via constructor and registered in a composer, allowing for customization.
Reverts 
https://github.com/umbraco/Umbraco-CMS/pull/9803 @nul800sebastiaan 

Fixes #9615

### CODE REVIEW

File: src/Umbraco.Web/Umbraco.Web.csproj [hunk 1]
```diff
@@ -1293,7 +1296,7 @@
     </PropertyGroup>
     <ItemGroup>
       <!-- we want to exclude all facade references ?! -->
-      <FixedReferencePath Include="@(ReferencePath)" Condition="'%(ReferencePath.FileName)' != 'System.ValueTuple' and '%(ReferencePath.FileName)' != 'System.Net.Http'" />
+      <FixedReferencePath Include="@(ReferencePath)" Condition="'%(ReferencePath.FileName)' != 'System.ValueTuple' and '%(ReferencePath.FileName)' != 'System.Net.Http' and '%(ReferencePath.FileName)' != 'System.Text.Encoding.CodePages'" />
```
Comment (nzdev) [inline src/Umbraco.Web/Umbraco.Web.csproj:1299]: This was the build fix

File: src/Umbraco.Web/Services/IconService.cs [hunk 2]
```diff
@@ -12,31 +13,24 @@ namespace Umbraco.Web.Services
     public class IconService : IIconService
     {
         private readonly IGlobalSettings _globalSettings;
+        private readonly IHtmlSanitizer _htmlSanitizer;
 
-        public IconService(IGlobalSettings globalSettings)
+        public IconService(IGlobalSettings globalSettings,IHtmlSanitizer htmlSanitizer)
```
Comment (nul800sebastiaan) [inline src/Umbraco.Web/Services/IconService.cs:18]: public IconService(IGlobalSettings globalSettings,IHtmlSanitizer htmlSanitizer)
```suggestion
        public IconService(IGlobalSettings globalSettings, IHtmlSanitizer htmlSanitizer)
```

Comment (nul800sebastiaan) [reply src/Umbraco.Web/Services/IconService.cs:18]: I know it's nitpicking! But it hurts my eyes.. ;-)

File: src/Umbraco.Web/Umbraco.Web.csproj (hunk 1 above)
Comment (nul800sebastiaan) [reply src/Umbraco.Web/Umbraco.Web.csproj:1299]: Interesting.. I am not sure what the problem we're trying to solve with `FixedReferencePath` is but.. if it works, it works 🤷

**Review by nul800sebastiaan — APPROVED** [review]

**Pull request merged**